pkg/api/nextdate.go

```diff
@@ -0,0 +1,100 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func NextDate(now time.Time, dstart string, repeat string) (string, error) {
+	if repeat == "" {
+		return "", fmt.Errorf("пустое правило повторения")
+	}
+
+	// Парсим дату начала
+	date, err := time.Parse(DateFormat, dstart)
+	if err != nil {
+		return "", fmt.Errorf("некорректная дата начала: %v", err)
```

Comment:
При оборачивании нужно использовать `%w` (wrap), а просто при выводе в консоль `%v`
```suggestion
		return "", fmt.Errorf("некорректная дата начала: %w", err)
```